### PR TITLE
chore: fix expr macro doc

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/common/src/types/mod.rs [hunk 1]
```diff
@@ -948,7 +958,7 @@ impl ScalarImpl {
                 }
                 let mut fields = Vec::with_capacity(s.len());
                 for (s, ty) in str[1..str.len() - 1].split(',').zip_eq_debug(s.types()) {
-                    fields.push(Some(Self::from_text(s.trim().as_bytes(), ty)?));
+                    fields.push(Self::from_pretty(s.trim(), ty)?);
```
Comment (wangrunji0408) [inline src/common/src/types/mod.rs:961]: `from_pretty` is only used for internal testing, so we introduced `.` and `(empty)` for convenience. But `from_text` is used for parsing literals, which is user-facing. I don't think it should depend on `from_pretty` here. Maybe we can follow the [literal syntax for struct values](https://www.postgresql.org/docs/current/rowtypes.html#ROWTYPES-CONSTRUCTING). It uses `` for NULL and `""` for empty string. e.g.
```
<I,I>
(,1)
(1,)
(,)
```

Comment (wangrunji0408) [reply src/common/src/types/mod.rs:961]: Btw, what is the difference between `ScalarImpl::from_literal` and `ScalarImpl::from_text`? Can we merge them now? cc @xiangjinwu

Comment (xiangjinwu) [reply src/common/src/types/mod.rs:961]: Yes they should be merged.

Comment (ZENOTME) [reply src/common/src/types/mod.rs:961]: > But from_text is used for parsing literals, which is user-facing. I don't think it should depend on from_pretty here. Maybe we can follow the literal syntax for struct values. It uses `` for NULL and "" for empty string.

Oh yes, my bad.

**Comment by stdrc** [comment]
Any updates?

**Comment by ZENOTME** [comment]
> Any updates?

I think it is completed now. struct part has been fixed in  #14497.

**Review by wangrunji0408 — APPROVED** [review]

**Pull request merged**